backend/src/test/java/codezap/template/service/facade/CategoryTemplateApplicationServiceTest.java

```diff
@@ -0,0 +1,131 @@
+    class CreateTemplate {
+
+        @Test
+        @DisplayName("성공: 작성자의 권한 확인 후 템플릿 생성")
+        void createTemplate_Success() {
+            // given
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Category membersCategory = categoryRepository.save(new Category("Members", member));
+            List<CreateSourceCodeRequest> sourceCodeRequests = List.of(
+                    new CreateSourceCodeRequest("filename", "content", 1));
+            CreateTemplateRequest request = new CreateTemplateRequest(
+                    "Template", "Description", sourceCodeRequests, 1, membersCategory.getId(), List.of("태그"));
+
+            // when
+            Long result = categoryTemplateApplicationService.createTemplate(member, request);
+
+            // then
+            assertThat(result).isEqualTo(1L);
+        }
+
+        @Test
+        @DisplayName("실패: 카테고리에 대한 권한이 없는 경우")
+        void createTemplate_Fail_NoAuthorization() {
+            // given
+            Member ownerMember = memberRepository.save(MemberFixture.getFirstMember());
+            Member otherMember = memberRepository.save(MemberFixture.getSecondMember());
+            Category category = categoryRepository.save(new Category("Members", ownerMember));
+            List<CreateSourceCodeRequest> sourceCodeRequests = List.of(
+                    new CreateSourceCodeRequest("filename", "content", 1));
+            CreateTemplateRequest request = new CreateTemplateRequest(
+                    "Template", "Description", sourceCodeRequests, 1, category.getId(), List.of("태그"));
+
+            // when & then
+            assertThatThrownBy(() -> categoryTemplateApplicationService.createTemplate(otherMember, request))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("해당 카테고리에 대한 권한이 없습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("템플릿 수정")
+    class UpdateTemplate {
+
+        @Test
+        @DisplayName("성공")
+        void updateTemplate_Success() {
+            // given
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Category category = categoryRepository.save(new Category("Members", member));
+            Template template = templateRepository.save(TemplateFixture.get(member, category));
+            SourceCode sourceCode = sourceCodeRepository.save(new SourceCode(template, "filename", "content", 1));
+            thumbnailRepository.save(new Thumbnail(template, sourceCode));
+            UpdateSourceCodeRequest updateSourceCodeRequest = new UpdateSourceCodeRequest(
+                    sourceCode.getId(), "Updated filename", "Updated content", 1);
+            UpdateTemplateRequest request = new UpdateTemplateRequest("Updated Template", "Updated Description", Collections.emptyList(),
+                    List.of(updateSourceCodeRequest), Collections.emptyList(), category.getId(), Collections.emptyList());
```

Comment:
개행을 하면 가독성이 더 좋아질 것 같아요
```suggestion
            UpdateTemplateRequest request = new UpdateTemplateRequest(
                  "Updated Template", 
                  "Updated Description", 
                  Collections.emptyList(),
                  List.of(updateSourceCodeRequest), 
                  Collections.emptyList(), 
                  category.getId(), 
                  Collections.emptyList());
```